source/System.cpp

```diff
@@ -526,6 +530,8 @@ void System::UpdateSystem(const Set<System> &systems, const set<double> &neighbo
 		minimumFleetPeriod = min<int>(minimumFleetPeriod, event.Period());
 	if(minimumFleetPeriod == numeric_limits<int>::max())
 		minimumFleetPeriod = 0;
+
+	actualLingerTime = requestedLingerTime >= 0 ? requestedLingerTime : minimumFleetPeriod / 4;
```

Comment:
Does `requestedLingerTime` even need to exist at all? It appears to only ever be used in this method and the unused getter method.
It doesn't seem to need to be a class member. The other method that refers to it is never used and so doesn't need to exist.
This method does use it, but it doesn't seem to need to.
Why not just have a single `lingerTime` variable with the default value being `-1`.
Then, if no value is given for it, it will still be negative and you can do this same thing just with only that variable.
```suggestion

	if(lingerTime < 0)
	lingerTime = minimumFleetPeriod / 4;
```

Reply:
That wouldn't work. If the minimum fleet arrival period changed, the lingerTime wouldn't be updated since it's not -1 anymore.

Consider this:

1. A system exists.
2. Add "linger time" to the system.
3. Linger time is now 1/4 of the minimum fleet arrival period.
4. Change the fleets in the system.
5. Linger time isn't 1/4 of the minimum fleet arrival period anymore because System forgot the -1 value.

The separate actualLingerTime and requestedLingerTime allows the System to remember it should use the 1/4 arrival period value, and automatically update that when the minimum arrival period changes.